llvm/lib/CodeGen/MachineSizeOpts.cpp

```diff
@@ -44,7 +48,9 @@ bool llvm::shouldOptimizeForSize(const MachineBasicBlock *MBB,
                                  MBFIWrapper *MBFIW,
                                  PGSOQueryType QueryType) {
   assert(MBB);
-  if (!PSI || !MBFIW)
+  if (MBB->getParent()->getFunction().hasOptSize())
+    return true;
+  if (!MBFIW)
```

Comment:
Was dropping `!PSI` here intentional?

Reply:
Yes because `PSI` is checked in `shouldOptimizeForSizeImpl()` anyway.

https://github.com/llvm/llvm-project/blob/8417f6af54c8f6dcf5893ab1352b50bf33c5a1ba/llvm/include/llvm/Transforms/Utils/SizeOpts.h#L72-L76